File: lib/backend/models/user_model.js
const mongoose = require("mongoose");

const userSchema = mongoose.Schema(
    {
       name: {
        type: String,
        required: true,
        trim: true
       },
       uid: {
        type: String,
        required: true,
        trim: true,
       },
       role: {
        type: String,
        default: "user",
        trim: true,
       },
       email: {
        type: String,
        required: true,
        trim: true,
       },
       password: {
        type: String,
        required: true,
        trim: true,
       },
       phone: {
        type: String,
        default: "",
        trim: true,
       }
    }
);

authModel = mongoose.model("model", userSchema);

module.exports = authModel;

File: lib/models/user_model.dart
import 'package:flutter/material.dart';

class UserModel {
  String name, token, email, id, uid, role, phone;

  UserModel({
    required this.email,
    required this.uid,
    required this.name,
    required this.token,
    required this.phone,
    required this.id,
    required this.role,
  });

}
